com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/MethodInvocationLocator.java

```diff
@@ -71,7 +71,7 @@ public boolean visit(MethodDeclaration node) {
 
     @Override
     public boolean visit(TypeDeclaration node) {
-        return shouldVisitNode(node);
+        return unit.getRoot() == node.getRoot() || shouldVisitNode(node);
```

Comment:
The condition check `unit.getRoot() == node.getRoot()` is not needed. I tested #519 snippet without this check, it worked.

Reply:
For some reason for me it start to fail before in `shouldVisitNode` the `end` becomes -1 since the on JDT side the start + length of the node becomes equal to the end of the compilation unit's last position. I will try again without this check.

Reply:
OK now I found the reason, if you don't have a empty line at the end of the class, that after the last class definitions' brace, then the end position becomes -1

Reply:
I can reproduce it, but i would prefer to modify the implementation `shouldVisitNode()` to fix the problem.

Since the offset range of the ast node is `[node.getStartPosition, node.getStartPosition() + node.getLength() - 1]`, the ending line should be got from `unit.getLineNumber(node.getStartPosition() + node.getLength() - 1)`.

The fix could be as follow:
```java
    private boolean shouldVisitNode(ASTNode node) {
        int start = unit.getLineNumber(node.getStartPosition());
        int end = unit.getLineNumber(node.getStartPosition() + node.getLength() - 1);

        if (line >= start && line <= end) {
            return true;
        }

        return false;
    }
```